File: Tarea 2/triangle.cpp
/*
    Arboles y Grafos 2025-1
    Santiago Guevara Idarraga
    Problem B: triangle
    Febrero 10 de 2025
*/

#include <iostream>
#include <cmath>
#include <iomanip>

using namespace std;

long double solve(long double AB, long double ratio) {
    long double low = 0, high = AB, mid, c = AB*pow(ratio / (1.0L + ratio), 1/2);
    while(high-low > 1e-6) {
        mid = (low + high)/2;
        if(mid < c) {
            low = mid;
        }
        else {
            high = mid;
        }
    }
    cout << fixed << setprecision(6) << AB*pow(ratio / (1.0L + ratio), 1/2) << endl;
    return mid;
}
int main() {
    int T, AB, AC, BC, ratio, k = 1;
    cin >> T;
    while(T--) {
        cin >> AB >> AC >> BC >> ratio;
        cout << fixed << setprecision(6) << solve(AB, ratio) << endl;
        k++;
    }
    return 0;
}

File: Tarea 2/triangle.py
"""
    Arboles y Grafos 2025-1
    Santiago Guevara Idarraga
    Problem B: triangle
    Febrero 13 de 2025
"""

from sys import stdin
from math import sqrt

def solve(ab, ratio):
    low, high = 0, ab
    while high-low > 10**-10:
        mid = (low+high)/2
        r = mid**2 / (ab**2 - mid**2)
        if r < ratio:
            low = mid
        else:
            high = mid
    return mid
def main():
    cases = int(stdin.readline())
    currCase = 1
    while cases:
        ab, ac, bc, ratio = map(float, stdin.readline().split())
        print(f"Case {currCase}: {solve(ab, ratio):.6f}")
        currCase += 1
        cases -= 1
main()